plugins/main/bridge/bridge.go

```diff
@@ -242,8 +253,21 @@ func cmdAdd(args *skel.CmdArgs) error {
 		return err
 	}
 
+	// Determine the plugin type
+	// TODO: This assumes that for dual-stack, the plugin type is the
+	// same for IPv4 vs. IPv6. To allow for different plugin types,
+	// the IPv4 and IPv6 IPAM config will need to be split up and
+	// passed to each plugin separately, with results being combined.
+	var plugin_type string
+	switch {
+	case n.IPAM.Type != "":
+		plugin_type = n.IPAM.Type
+	default:
+		plugin_type = n.IPAM6.Type
+	}
```

Comment:
To add more detail... to handle the case where the config calls for a different IPAM plugin for V4 vs. V6, we would have to take the IPAM / IPAM6 config objects, separately marshal these objects into stdin (byte stream) format, call the separate IPAM plugins, and then merge the results. In the simpler case where the V4 and V6 use the same IPAM plugin, the net config stdin can be used as is. It's not terribly complicated, but I was going to add this in later.